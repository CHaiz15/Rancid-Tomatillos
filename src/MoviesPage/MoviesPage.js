import React from 'react';
import MovieThumbnail from '../MovieThumbnail/MovieThumbnail.js'

const MoviesPage = () => {
  return (
    <section>
      <h1>Movies...</h1>
      <MovieThumbnail />
    </section>
  )
}

export default MoviesPage;