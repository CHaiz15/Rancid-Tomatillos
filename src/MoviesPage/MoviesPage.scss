.movies-page {
  width: 100%;
}

.film-block {
  background: url('../assets/film-bg.jpg')no-repeat center center;
  background-size: cover;
  background-attachment: fixed;
  object-fit: contain;
  width: 100%;
  height: 30rem;
  border: 1px solid black;
}

.movies-container {
  margin: 2% 0 0 0;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}