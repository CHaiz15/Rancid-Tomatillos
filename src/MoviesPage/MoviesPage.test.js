import React from 'react';
import { MoviesPage, mapStateToProps } from './MoviesPage';
import { shallow } from 'enzyme';

describe('MoviesPage', () => {
  let movies
  it('should match the snapshot', () => {
    movies = [{
      id: 21,
      title: "Sonic the Hedgehog",
      "poster_path": "https://image.tmdb.org/t/p/original//aQvJ5WPzZgYVDrxLX4R6cLJCEaQ.jpg",
      "backdrop_path": "https://image.tmdb.org/t/p/original//tCUcf3oNWMW8kwAj3WC6CvIN5ah.jpg",
      "release_date": "2020-02-12",
      "overview": "Based on the global blockbuster videogame franchise from Sega, Sonic the Hedgehog tells the story of the world’s speediest hedgehog as he embraces his new home on Earth. In this live-action adventure comedy, Sonic and his new best friend team up to defend the planet from the evil genius Dr. Robotnik and his plans for world domination.",
      "average_rating": 8
  }];
    const wrapper = shallow(<MoviesPage movies={movies}/>);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with an array of movies', () => {
      const mockState = {
        movies: [{
          id: 21,
          title: "Sonic the Hedgehog",
          "poster_path": "https://image.tmdb.org/t/p/original//aQvJ5WPzZgYVDrxLX4R6cLJCEaQ.jpg",
          "backdrop_path": "https://image.tmdb.org/t/p/original//tCUcf3oNWMW8kwAj3WC6CvIN5ah.jpg",
          "release_date": "2020-02-12",
          "overview": "Based on the global blockbuster videogame franchise from Sega, Sonic the Hedgehog tells the story of the world’s speediest hedgehog as he embraces his new home on Earth. In this live-action adventure comedy, Sonic and his new best friend team up to defend the planet from the evil genius Dr. Robotnik and his plans for world domination.",
          "average_rating": 8
      }],
      }

      const expected = {
        movies: [{
          id: 21,
          title: "Sonic the Hedgehog",
          "poster_path": "https://image.tmdb.org/t/p/original//aQvJ5WPzZgYVDrxLX4R6cLJCEaQ.jpg",
          "backdrop_path": "https://image.tmdb.org/t/p/original//tCUcf3oNWMW8kwAj3WC6CvIN5ah.jpg",
          "release_date": "2020-02-12",
          "overview": "Based on the global blockbuster videogame franchise from Sega, Sonic the Hedgehog tells the story of the world’s speediest hedgehog as he embraces his new home on Earth. In this live-action adventure comedy, Sonic and his new best friend team up to defend the planet from the evil genius Dr. Robotnik and his plans for world domination.",
          "average_rating": 8
      }]
      }
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })
  })
});