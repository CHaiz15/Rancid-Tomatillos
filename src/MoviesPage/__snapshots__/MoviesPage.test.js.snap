// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesPage should match the snapshot 1`] = `
<section
  className="movies-page"
>
  <div
    className="film-block"
  />
  <section
    className="movies-container"
  >
    <Link
      className="movie-thumbnail"
      to="/movies/21"
    >
      <MovieThumbnail
        average_rating={8}
        backdrop_path="https://image.tmdb.org/t/p/original//tCUcf3oNWMW8kwAj3WC6CvIN5ah.jpg"
        id={21}
        overview="Based on the global blockbuster videogame franchise from Sega, Sonic the Hedgehog tells the story of the worldâ€™s speediest hedgehog as he embraces his new home on Earth. In this live-action adventure comedy, Sonic and his new best friend team up to defend the planet from the evil genius Dr. Robotnik and his plans for world domination."
        poster_path="https://image.tmdb.org/t/p/original//aQvJ5WPzZgYVDrxLX4R6cLJCEaQ.jpg"
        release_date="2020-02-12"
        title="Sonic the Hedgehog"
      />
    </Link>
  </section>
</section>
`;
